src/Reference/ContainsOne.php

```diff
@@ -98,8 +98,8 @@ public function ref($defaults = []): Model
         // set some hooks for ref_model
         foreach ([Model::HOOK_AFTER_SAVE, Model::HOOK_AFTER_DELETE] as $spot) {
             $theirModel->onHook($spot, function ($theirModel) {
-                $row = $theirModel->persistence->getRawDataByTable($this->table_alias);
-                $row = $row ? array_shift($row) : null; // get first and only one record from array persistence
+                $rows = iterator_to_array($theirModel->persistence->getRawDataIterator($this->table_alias));
+                $row = $rows ? array_shift($rows) : null; // get first and only one record from array persistence
```

Comment:
is this correct when more than 1 result is returned?

when something is 1:1, only one result should be returned, otherwise throw, right?

Reply:
I modified this now to how it should be - using Query